devicetypes/stelpro/stelpro-ki-zigbee-thermostat.src/stelpro-ki-zigbee-thermostat.groovy

```diff
@@ -362,11 +353,32 @@ def handleTemperature(descMap) {
 			log.debug "CLEAR ALARM @ $map.value $map.unit (raw $intVal)"
 			sendEvent(name: "temperatureAlarm", value: "cleared")
 		}
+
+		handleOperatingStateBugfix(null, map.value)
```

Comment:
you could tidy up the function below if you just passed in the current values here instead of getting them down there

Then you could avoid the extra variables you're creating.

Reply:
It sounds like given some testing results, and the behavior of this bug, I might go about this a different way. More after my testing.